Describe replication_connect_quorum meaning (#23)
tarantool/tarantool-internals: source/replication.rst

            ...
        }

> There are some differences in how many connections are considered enough to

Reviewer: That's quite long wall of text. Can we add an extra heading here?

Author: I would then need to add a heading for the remaining part of the text describing how appliers are connected, and I couldn't find a suitable name for it, because it's the same logical piece about replicaset_connect.

I added extra separators instead:
```diff
diff --git a/source/replication.rst b/source/replication.rst
index 3bb3165..1d726d3 100644
--- a/source/replication.rst
+++ b/source/replication.rst
@@ -174,6 +174,8 @@ of appliers in connected state.
             ...
         }
 
+----
+
 There are some differences in how many connections are considered enough to
 proceed. This is controlled by ``bool connect_quorum`` parameter of
 ``replicaset_connect``. On a instance bootstrap, i.e. when there are no local
@@ -215,6 +217,8 @@ passes ``replicaset_connect`` faster:
             if (state.connected >= quorum && !connect_quorum)
                 break;
 
+----
+
 Once appliers are in ``APPLIER_CONNECTED`` state we clear ``applier_on_connect_f``
 trigger and call ``replicaset_update``. Note that not all appliers might
 be connected. Those ones which did not manage to are explicitly stopped,
```

proceed. This is controlled by ``bool connect_quorum`` parameter of
``replicaset_connect``. On a instance bootstrap, i.e. when there are no local
files (.snap, .xlog) to recover from ``connect_quorum`` is ``true``, and this
means that ``replicaset_connect`` tries to connect to ``count`` of remote nodes
for a period of ``replication_connect_timeout``. If the instance fails to
connect to ``count`` of nodes, it continues with at least
``replication_connect_quorum`` of connections, otherwise, an error is thrown:

> .. code-block:: c

Reviewer: ```suggestion
..  code-block:: c
```

    replicaset_connect
        ...
        while (state.connected < count)
            ...
        if (state.connected < count)
            ...
            if (connect_quorum && state_connected < quorum) {
                error
                ...

> This is important to try to connect to everyone during bootstrap, because

Reviewer: Too much `to` in one row. 
```suggestion
It is important to try connecting to everyone during bootstrap, because
```

connected appliers are later polled to find the so-called "bootstrap leader",
the node, which will add everyone else to the replicaset. Having a couple of
nodes fail to connect to everyone means they will operate on differing sets of
connections when finding the bootstrap leader, which essentially leads to
conflicts, when two nodes are registering replicas at the same time.

On a normal start though, connecting to everyone is not so critical.
That's why ``connect_quorum`` is ``false`` in this case. This means the node
passes ``replicaset_connect`` faster:

> .. code-block:: c

Reviewer: ```suggestion
..  code-block:: c
```

Note that woken appliers are not running they are just marked
as alive.

After waking up all appliers comes the ``sync`` stage:

> .. code-block:: c

Reviewer: ```suggestion
..  code-block:: c
```
